pkgs/games/mudlet/default.nix

```diff
@@ -1,46 +1,68 @@
-{ fetchurl, unzip, stdenv, makeWrapper, qtbase, yajl, libzip, hunspell
-, boost, lua5_1, luafilesystem, luazip, lrexlib-pcre, luasql-sqlite3, qmake }:
+{ fetchFromGitHub, fetchpatch, stdenv, qt5, pcre, pugixml, qtbase, qtmultimedia, qttools, yajl, libzip, hunspell
+, boost, libGLU, lua51Packages, cmake,  which, }:
 
+let
+  luaVersion = "5.1";
+  luaEnv = lua51Packages.lua.withPackages(ps: with ps; [ luazip luafilesystem lrexlib-pcre luasql-sqlite3 lua-yajl luautf8 ]);
+in
 stdenv.mkDerivation rec {
   pname = "mudlet";
-  version = "3.0.0-delta";
+  version = "4.0.3";
 
-  src = fetchurl {
-    url = "https://github.com/Mudlet/Mudlet/archive/Mudlet-${version}.tar.gz";
-    sha256 = "08fhqd323kgz5s17ac5z9dhkjxcmwvcmvhzy0x1vw4rayhijfrd7";
+  src = fetchFromGitHub {
+    owner = "Mudlet";
+    repo = "Mudlet";
+    rev = "Mudlet-${version}";
+    fetchSubmodules = true;
+    sha256 = "18bl4k0qgh47d9k5ipfvypfj1il678c0ws64a8adn8k21jajzkik";
   };
 
-  nativeBuildInputs = [ makeWrapper qmake ];
+  patches = [
+    ( fetchpatch {
+      url = "https://github.com/abbradar/Mudlet/commit/4f9450e9f66ac72f502e52951ff04b8ac8d123b3.patch";
+      name = "hunspell-1.7";
+      sha256 = "09qggls4pzpd8h9h10fbpfd7x3kr7fjp9axdwz98igpwy714n98j";
+    })
+  ];
+
+  nativeBuildInputs = [ cmake qt5.wrapQtAppsHook qttools which ];
   buildInputs = [
-    unzip qtbase lua5_1 hunspell libzip yajl boost
-    luafilesystem luazip lrexlib-pcre luasql-sqlite3
+    pcre pugixml qtbase qtmultimedia luaEnv libzip libGLU yajl boost hunspell
   ];
 
-  preConfigure = "cd src";
+  WITH_FONTS = false;
+  WITH_UPDATER = "NO";
+
+  preConfigure = ''
```

Comment:
Perhaps this is not needed now that we use `withPackages`?

Reply:
`preConfigure` removed.